### ISSUE TITLE
Support folding blocks

### ISSUE THREAD

**Issue by paultyng** [hashicorp/terraform-ls#105]
https://microsoft.github.io/language-server-protocol/specification#textDocument_foldingRange

**Comment by amasover** [comment]
Apologies as I'm not too familiar with the LSP spec. Would this allow collapsing/expanding blocks, e.g., resource blocks, etc? If so, pretty cool!

**Comment by paultyng** [comment]
@amasover yep, exactly. I'd imagine the simplistic initial implementation will just be folding of root level blocks (resource, provider, data, variable, output, module, local, terraform, etc.).

**Comment by PePoDev** [comment]
@paultyng  I have try with this, But not working.
```hcl
#region test
variable "test" {
  type        = bool
  default     = true
}
#endregion
```

**Comment by jpogran** [comment]
@PePoDev we haven't implemented region folding yet. When we do you'll see a PR and a line item in our changelog

**Comment by aleqsss** [comment]
Hi,

I'm also interested in region folding, any news in regards of this? 🙂🤞

**Comment by dbanck** [comment]
Hi @aleqsss,

Sorry, but there hasn't been any progress on this. We are currently planning our roadmap for the first quarter, and it's already full of other issues we need to address. So at this point, it's tough for me to say when we can work on folding ranges.

**Comment by aleqsss** [comment]
Understood, and thank you for the information!

**Comment by garsiv1932** [comment]
Is it too difficult?

**Comment by jpogran** [comment]
Hey @garsiv1932 , just to clarify region folding support was implemented in the Terraform VS Code extension in https://github.com/hashicorp/vscode-terraform/pull/924, so if you do want to use them now we do support them. If you use another editor, that has not been planned yet.

**Comment by garsiv1932** [comment]
> Hey @garsiv1932 , just to clarify region folding support was implemented in the Terraform VS Code extension in https://github.com/hashicorp/vscode-terraform/pull/924, so if you do want to use them now we do support them. If you use another editor, that has not been planned yet.

And how is the syntax, because i tried to put # regiion and #endRegion and didnt work.

Can You post the link of the vscode plugin too, only yo clarify if i AM using the dame one.

**Comment by garsiv1932** [comment]
> And how is the syntax, because i tried to put # regiion and #endRegion and didnt work.
> 
> Can You post the link of the vscode plugin too, only yo clarify if i AM using the dame one.

And how is the syntax, because i tried to put # regiion and #endRegion and didnt work.

Can You post the link of the vscode plugin too, only yo clarify if i AM using the dame one.

**Comment by garsiv1932** [comment]
> Hey @garsiv1932 , just to clarify region folding support was implemented in the Terraform VS Code extension in [hashicorp/vscode-terraform#924](https://github.com/hashicorp/vscode-terraform/pull/924), so if you do want to use them now we do support them. If you use another editor, that has not been planned yet.

And how is the syntax, because i tried to put # regiion and #endRegion and didnt work.

Can You post the link of the vscode plugin too, only yo clarify if i AM using the dame one.

**Comment by garsiv1932** [comment]
I am checking and it is not working correctly on my machine, I don't why, I closed my code, and reopened it, and the arrow showed for one second. I created some regions on my code and the arrow is not showing, the only way to make it work to close vscode, and when I reopen, when the arrow shows by one second, with all my speed, click on the arrow, only in that way vscode remember the regions on my code.

I uninstalled all the unused extensions on my vscode, but the problem persist.

**Comment by garsiv1932** [comment]
I had a conflict on my extensions, is working right now.

**Comment by artemudovyk** [comment]
> The syntax is `#region` and `#endregion`:

Sorry for nitpicking, but why is `# region` and `# endregion` not supported?
![image](https://github.com/user-attachments/assets/faf727d1-e09b-4111-af9f-901454b0904a)

**Comment by rcjsuen** [comment]
> > The syntax is `#region` and `#endregion`:
> 
> Sorry for nitpicking, but why are version with whitespace `# region` and `# endregion` not supported?

@artemudovyk Is this still a problem for you? It seems to be working for me 

<img width="458" alt="image" src="https://github.com/user-attachments/assets/25271e52-2e06-4981-a2fd-0a315359ec08">

**Comment by artemudovyk** [comment]
> @artemudovyk Is this still a problem for you? It seems to be working for me

Yeah, it's still the same for me.

**Comment by rcjsuen** [comment]
> Yeah, it's still the same for me.

I see the difference now. I was using v0.2.5 of [4ops.terraform](https://marketplace.visualstudio.com/items?itemName=4ops.terraform). Indeed, if I use v2.32.3 of [HashiCorp.terraform](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform) then it doesn't work. 👍
